tests/backends/sqlite/tests.py

```diff
@@ -245,3 +246,55 @@ def create_object():
             for conn in thread_connections:
                 if conn is not main_connection:
                     conn.close()
+
+
+@unittest.skipUnless(connection.vendor == "sqlite", "SQLite tests")
+@override_settings(DEBUG=True)
+class TestTransactionMode(TransactionTestCase):
+    available_apps = ["backends"]
+
+    def test_default_transaction_mode(self):
+        with transaction.atomic():
+            pass
+        self.assertEqual(connection.queries[0]["sql"], "BEGIN")
+
+    def test_invalid_transaction_mode(self):
+        connection = self._get_connection_for_transaction_mode("invalid")
+        with self.assertRaises(ImproperlyConfigured):
+            connection.ensure_connection()
+
+    def test_valid_transaction_modes(self):
+        valid_transaction_modes = ("deferred", "immediate", "exclusive")
+        for transaction_mode in valid_transaction_modes:
+            with self.subTest(transaction_mode):
+                connection = self._get_connection_for_transaction_mode(transaction_mode)
+                connection.ensure_connection()
+                with transaction.atomic():
+                    pass
+
+                self.assertEqual(
+                    connection.queries[0]["sql"], f"BEGIN {transaction_mode.upper()}"
+                )
+
+    def _get_connection_for_transaction_mode(self, transaction_mode):
+        settings_dict = {
+            "default": {
+                "ENGINE": "django.db.backends.sqlite3",
+                "NAME": ":memory:",
+                "TEST": {"NAME": ":memory:"},
+                "OPTIONS": {
+                    "transaction_mode": transaction_mode,
+                },
+            }
+        }
+        connection = ConnectionHandler(settings_dict)["default"]
+        self.addCleanup(connection.close)
```

Comment:
I'd suggest `overried_settings` and accessing `connections[alias_created_for_tests]` instead of dealing with the low level connection handler machinery.